mpyg321/mpyg321.py

```diff
@@ -34,43 +34,28 @@
     },
     {
         "mpg_code": "@silence",
-        "action": "user_silence",
+        "action": None,
         "description": "Player has been silenced by the user."
     },
     {
-        "mpg_code": "@V *",
-        "action": "volume_change",
-        "description": "Volume change event."
-    },
-    {
-        "mpg_code": "@VOLUME *",
-        "action": "volume_change",
-        "description": "Volume change event."
-    },
-    {
-        "mpg_code": "@LOADLIST *",
-        "action": "load_list",
-        "description": "Load list event."
-    },
-    {
-        "mpg_code": "@LL *",
-        "action": "load_list",
-        "description": "Load list event."
+        "mpg_code": "@V [0-9\.\s%]*",
+        "action": None,
+        "description": "Volume change event.",
     },
     {
-        "mpg_code": "@S *",
-        "action": "pass",
-        "description": "Ignore event."
+        "mpg_code": "@S [a-zA-Z0-9\.\s-]*",
+        "action": None,
+        "description": "Stereo info event."
     },
     {
-        "mpg_code": "\r\n",
-        "action": "pass",
-        "description": "Ignore event."
+        "mpg_code": "@I *",
+        "action": None,
+        "description": "Information event."
     },
     {
-        "mpg_code": "",
-        "action": "pass",
-        "description": "Ignore event."
+        "mpg_code": pexpect.TIMEOUT,
```

Comment:
With that, pexpect continue to work properly. If I don't handle it, pexpect goes unstable and output errors on screen;

Reply:
I'm not sure I understand the problem you had... Is it that after some time without any actual event, pexpect has issues? How long? How can I reproduce the problem you faced?

Reply:
Well this happen when I play a mp3 file with mpg123; mpg123 outpout informations when it start a mp3 and if we don't handle all of the ouput, it will throw a TIMEOUT.

In my use case, after 30 seconds (the default delay) pexpect throws a TIMEOUT but I don't know what cause the TIMEOUT because the property childProcess.Before is None, so I don't know what was expected, that's a weird problem.

To know when it goes in TIMEOUT, it's possible in the process_output method and add an if action == 'timeout' for instance, that's the place to check the .Before and .After properties.

Also, I think adding pexpect.EOF in the same way to handle the EOF issue properly in #10

Finally, when you check pexpect documentation and examples, most of the programmers handle pexpect.TIMEOUT and pexpect.EOF to keep thinks stable in their program. Actually in the lib, we ignore TIMEOUT and EOF and I think it's a problem.

https://www.programcreek.com/python/example/8691/pexpect.TIMEOUT

My strategy is : if a timeout occurs, I handle it but I dont care about it.

What do you think ?

Thanks !

Reply:
I've done some testing on my mac, and I wasn't able to reproduce the crash with the timeout, so I'll trust you on that :-)